primitives/src/contract_metadata.rs

```diff
@@ -214,4 +214,8 @@ impl ContractMetadata {
             license,
         }
     }
+
+    pub fn get_contract_type(&self) -> ContractType {
+        ContractType::decode(&mut self.contract_type.as_slice()).unwrap()
```

Comment:
We want to have a nicer way of doing this without unwrapping incase the store gets mangled. Treat panics as big problemo. Since panicking isnt safe just because the thread gets spawned

Reply:
Ok, I'll refactor